samgeo/samgeo3.py

```
@@ -4148,6 +4160,42 @@ def __init__(
+            except Exception:
+                pass
+            finally:
+                SamGeo3Video._active_predictor = None
+
+        # Fallback: a process group may still be initialized (e.g. it was set up
+        # outside of SamGeo3Video, or the previous shutdown failed part-way).
+        if torch.distributed.is_available() and torch.distributed.is_initialized():
+            try:
+                torch.distributed.destroy_process_group()
+            except Exception:
+                pass
```

Ruff (0.15.20)

[warning] Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(Exception): ...`

(SIM105)